src/main/java/com/hierynomus/smbj/Config.java

```diff
@@ -39,6 +40,7 @@
     private int readBufferSize;
     private int writeBufferSize;
     private int transactBufferSize;
+    private TransportMode transportMode;
```

Comment:
Wouldn't it be nicer to inject the TransportFactory here immediately, instead of the enum?